aips/aip-103.md

```diff
@@ -0,0 +1,110 @@
+
+Many customers of Auth0 and Cognito have applications with different branding within the same user-pool/tenant ecosystem.  Thus it is natural for such customers to use different application identifiers for their applications for organizational purposes.  This AIP will enable Aptos Federated Keyless Accounts to be interoperable across such applications.
+
+## Impact and risks
+
+- The relaxation of the `aud` field will allow for broader interoperability across applications within the same user-pool/tenant.  This allows for broader adoption of Aptos Federated Keyless Accounts in such user ecosystems.
+
+Risks
+- Developers need to not use `aud`-less accounts when not appropriate. This can be mitigated by the Aptos SDK default behavior so that developers must explicitly enable using `aud`-less accounts.
+- This introduces an additional proving path, one where the `aud` is not checked.  It is important that such proofs are rejected if the account requires `aud` to be present, as encoded in the KeylessPublicKey.
+- As circuit changes are needed to support `aud`-less accounts, a new ceremony will be needed to generate the proving key and verification key.
+- We want such accounts to be limited to Federated Keyless Accounts, as constructing Keyless Accounts without aud checks is unsafe.  This can be mitigated by the Aptos SDK disallowing `aud`-less accounts from being used as Keyless Accounts.  The prover will also reject proof requests for Keyless providers (as of now Google and Apple).  However, in a world where 3rd party provers are permitted, we cannot prevent developers from using `aud`-less accounts as Keyless Accounts, but developers would not have any incentive construct such accounts for their users (these accounts would be accessable by any other dApp, regardless of trust).
+- The verification key will need an update, which will invalidate all existing proofs.  Additionally the prover will need to start proving with the new proving key right away after the update.  The prover has already been updated to support the proving key rotations and the SDK also supports state checks to invalidate old proofs.
+
+## Alternative solutions
+
+The alternative is to add an additional keyless public key type where the formula to compute the IdCommitment does not contain the `aud` at all.
+
+This is the advantage of explicit type safety as a completely new validation path would be implemented.  There would be no risk of such proofs being accepted for accounts that require `aud` to be present due to explicit differences in how the proof would be validated gated on the type of public key.
+
+However the drawbacks include:
+- We need to add a new keyless public key type, which may not be needed if we can leverage the existing design.  And avoiding proliferation of keyless public key types is desirable.
```

Comment:
the proposed approach needs a flag in `KeylessPublicKey` to indicate "should skip aud", right?
that's unavoidably also a new keyless public key type?

Reply:
No. It would not.  skip aud would only be in the request to the prover and in the SDK to contruct a public key with aud being '' when calculating the IDC